Separate the installation of pre-commit to -f|--format case

AndrejOrsula opened the pull request:
# Description

This PR separates the installation of `pre-commit` from the main installation via `orbit.sh -i|--install` into the already existing `-f|--format` case.

This means that end-users are no longer required to install `pre-commit` when using the included `orbit.sh -i|--install` option, which should simplify the Dockerfile layers (#17). 

With this PR, `pre-commit` is automatically installed with `orbit.sh -f|--format` option if its command is not detected on the system. Furthermore, this PR makes sure that `pre-commit run` is always executed in the correct directory regardless of `${PWD}` when calling  `orbit.sh -f|--format` . Contributors are already informed to run pre-commit via the PR checklist found in the template (as seen below).

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## Checklist

- [x] I have run the [`pre-commit` checks](https://pre-commit.com/) with `pre-commit run --all-files` (see [here](https://pre-commit.com/#install) instructions to set it up)
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have updated the changelog and the corresponding version in the extension's `config/extension.toml` file

Mayankm96 commented on orbit.sh:
```diff
@@ -146,8 +144,15 @@ while [[ $# -gt 0 ]]; do
         -f|--format)
+            if ! command -v pre-commit &>/dev/null; then
+                echo "[INFO] Installing pre-commit..."
+                pip install pre-commit
```
Wondering if this should be:
```bash
python_exe=$(extract_isaacsim_python)
# install the rl-frameworks specified
${python_exe} -m pip install pre-commit
```

This way we don't install stuff to a user's local or global python site-packages?

AndrejOrsula replied:
Hmm, I am not sure. I will leave that up to you maintainers to decide.

One question, though. Would a contributor then need to extend `PATH`/`PYTHONPATH` (or source Isaac Sim python env) in order to `pre-commit run`? If so, it might complicate things too much.

Mayankm96 commented:
We should probably make doc changes at the following places to only use the bash utilities instead of pre-commit. What do you think?

* PR Checklist: https://github.com/NVIDIA-Omniverse/Orbit/blob/main/.github/PULL_REQUEST_TEMPLATE.md#checklist
* Developer's Guide: https://github.com/NVIDIA-Omniverse/Orbit/blob/main/docs/source/setup/developer.rst#code-style
* PR Guideline: https://github.com/NVIDIA-Omniverse/Orbit/blob/main/docs/source/refs/contributing.rst#coding-style

AndrejOrsula commented:
> We should probably make doc changes at the following places to only use the bash utilities instead of pre-commit. What do you think?
> 
> * PR Checklist: https://github.com/NVIDIA-Omniverse/Orbit/blob/main/.github/PULL_REQUEST_TEMPLATE.md#checklist
> * Developer's Guide: https://github.com/NVIDIA-Omniverse/Orbit/blob/main/docs/source/setup/developer.rst#code-style
> * PR Guideline: https://github.com/NVIDIA-Omniverse/Orbit/blob/main/docs/source/refs/contributing.rst#coding-style

I added instructions to use `./orbit.sh --format` instead of `pre-commit run --all-files` in those three files. I don't know if there's a better way of formatting it (I think that `./orbit.sh --format` is more informative than `./orbit.sh -f`). 
In one of the files, I also removed a link to the installation instructions of `pre-commit` as it is done automatically via  `./orbit.sh -f|--format` now.

Please, review this change (https://github.com/NVIDIA-Omniverse/Orbit/pull/28/commits/ceaf2a075661ed3bed336e0d7d0efcb145c0677a).
